### PR TITLE
AIP4 - constitution update for election delay

### PR DESCRIPTION
**Pull request by DZGoldman**

### CODE REVIEW

**Review by godzillaba — APPROVED** [review]

File: scripts/proposals/AIP3/verifyDeployment.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+dotenv.config();
+
+const { actionAddress } = JSON.parse(
+  fs.readFileSync("./scripts/proposals/AIP3/data/42161-AIP3-data.json").toString()
+);
+
+const ARB_URL = process.env.ARB_URL;
+if (!ARB_URL) throw new Error("ARB_URL required");
+const expectedAddressRegistry = "0x56C4E9Eb6c63aCDD19AeC2b1a00e4f0d7aBda9d3";
+const expectedConstitutionHash =
```
Comment (fredlacs) [inline scripts/proposals/AIP3/verifyDeployment.ts:15]: I get a different constitution hash in https://github.com/ArbitrumFoundation/docs/pull/597

Comment (fredlacs) [reply scripts/proposals/AIP3/verifyDeployment.ts:15]: nvm got the correct hash now

**Comment by fredlacs** [comment]
constitution hash in https://github.com/ArbitrumFoundation/docs/pull/598

File: src/gov-action-contracts/AIPs/AIP3Action.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "../address-registries/L2AddressRegistry.sol";
+
+/// @notice Governance action for constitution update relevant to security council elections, prior to their actual activation. See https://forum.arbitrum.foundation/t/proposal-update-security-council-election-start-date-to-ensure-time-for-security-audit/15426
+contract AIP3Action {
+    IL2AddressRegistry public immutable l2GovAddressRegistry;
+
+    bytes32 public constant newConstitutionHash = bytes32(0x70ae8e80709ba3edf810e4518056ce875a34254fe1138d7baa59d984f9372d71);
```
Comment (gzeoneth) [inline src/gov-action-contracts/AIPs/AIP3Action.sol:10]:
```suggestion
    bytes32 public constant newConstitutionHash =
        0x70ae8e80709ba3edf810e4518056ce875a34254fe1138d7baa59d984f9372d71;
```

**Review by gzeoneth — APPROVED** [review]

File: scripts/proposals/AIP4/data/421613-AIP4-data.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  "arbSysSendTxToL1Args": {
```
Comment (fredlacs) [inline scripts/proposals/AIP4/data/421613-AIP4-data.json:5]: would be cool if we could also include in this data json the calldata that is expected by the L2 Governor (like target address being ArbSys, and the calldata to it

Comment (fredlacs) [reply scripts/proposals/AIP4/data/421613-AIP4-data.json:5]: can make this change in a future PR tho

**Review by gzeoneth — APPROVED** [review]
lgtm

**Review by fredlacs — APPROVED** [review]
LGTM

**Comment by fredlacs** [comment]
AIP is now live onchain
https://www.tally.xyz/gov/arbitrum/proposal/94082931409036991607289748320100191193221627046136638381280146914481892861727

**Pull request merged**